src/dom.js

```diff
@@ -435,16 +410,55 @@ export function childElementByTag(parent, tagName) {
  * @return {!Array<!Element>}
  */
 export function childElementsByTag(parent, tagName) {
+  return scopedQuerySelectorAll(parent, `> ${tagName}`);
+}
+
+
+/**
+ * Finds the first element that matches `selector`, scoped inside `root`.
+ * Note: in IE, this causes a quick mutation of the element's class list.
+ * @param {!Element} root
+ * @param {string} selector
+ * @return {?Element}
+ */
+export function scopedQuerySelector(root, selector) {
   if (scopeSelectorSupported == null) {
-    scopeSelectorSupported = isScopeSelectorSupported(parent);
+    scopeSelectorSupported = isScopeSelectorSupported(root);
   }
   if (scopeSelectorSupported) {
-    return toArray(parent.querySelectorAll(':scope > ' + tagName));
+    return root./*OK*/querySelector(`:scope ${selector}`);
   }
-  tagName = tagName.toUpperCase();
-  return childElements(parent, el => {
-    return el.tagName == tagName;
-  });
+
+  // Only IE.
+  const unique = `i-amphtml-scoped-${Date.now()}`;
```

Comment:
It's removed immediately, so it'll still be unique. Really, all the class name needs is "i-amphtml-scoped".